### PR TITLE
Redesign

### PR DESCRIPTION
**Pull request by bytzo**
Fixed some layout elements and adds new styles. 

- Added support for mobile devices
- Changed button hover effect
- Changed navigation bar system to a simpler layout that is compatible with mobile devices
- Added the class `column` for use with the classes `col-1-3`, `col-2-3`, and `col-1-2` instead of `grid` which caused some issues.

You can preview the change here: [https://www.devision1088.com/unics.club/](https://www.devision1088.com/unics.club/)

**_EDIT: Link broken since my pages site is not hosted as root. Instead, clone and run on a local server. You will need git and python installed._**
```
$ git clone https://github.com/devision1088/unics.club.git
$ cd unics.club
$ python3 -m http.server
```
Run this after cloning in a seperate process:
```
$ open http://localhost:8000
```

**Before**
layour_error
**After**
fixed

### CODE REVIEW

**Comment by devision1088** [comment]
I fixed the width of the columns. Instead of being 32% wide to make up space for spaces rendered by HTML, the columns are exactly 33.333% wide. The spaces were deleted by updating the HTML to contain the tags directly next each other without line breaks in the tags. For example:

```
<div> ... </div>
<div> ... </div>
```
would be
```
<div> ... </div><div> ... </div>
```

**Comment by jason-s-yu** [comment]
@devision1088 Thanks for your changes. What is the reason behind moving each page to a folder and renaming them index.html?

**Comment by devision1088** [comment]
Thanks for your review!

The pages were moved to their own directory with index files to follow the current standards and practices. As seen in [the HTML 4 specification section 5.1.2 and 5.1.3 in 1997](https://www.w3.org/TR/WD-html40-970917/htmlweb.html), linking to an `.html` file was a common practice. Today, however, it is a common practice to link to directories because it allows for subpages and it follows [modern file path rules](https://www.w3schools.com/Html/html_filepaths.asp).

In the current website, the file paths linked to [https://unics.club/about](https://unics.club/about). Modern filesystems would try to find a directory called `about`, which does not exist. Instead, the proper way to link the about page would be [https://unics.club/about.html](https://unics.club/about.html), as it points to the actual file `/about.html`. However, we do not want the user to see `.html` as it adds to the complexity of the URL. Now, with the new file structure, we can properly link to [https://unics.club/about](https://unics.club/about) as it is a directory. This would then display `/about/index.html` because the index file is part of the about directory.

This change also allows additional sub-pages in the file structure, like [https://unics.club/about/people](https://unics.club/about), by putting directories in directories. For example, `/about/people/index.html` would be the URL [https://unics.club/about/people](https://unics.club/about). Hopefully this helps!

I now realize that there are some accessibility issues for people who need to use screen readers or switch controls. I'll commit again with the changes soon.

**Comment by devision1088** [comment]
**Note this is not ready for merging - still developing**
**This may need to be reverted because of the major changes**

I may have gone a little overboard on accessibility so this commit may need to be reverted since it is so different from the current design.

_Changes made/planned_
- [x] Improved full-width navigation that automatically adapts to large or small screens
- [x] New hero image (shown below)
- [x] Dark support
- [x] SEO
- [ ] Transfer content from old website
- [ ] Support screens smaller than 600px wide

_Used libraries_
- [three.js](https://threejs.org) for 3D hero image

Screen Shot 2019-11-11 at 9 25 03 PM
I have also added dark support for people who like dark mode.
Screen Shot 2019-11-11 at 9 28 11 PM

To preview (need python and git):
```
$ git clone https://github.com/devision1088/unics.club.git
$ cd unics.club
$ python3 -m http.server
```
```
$ open http://localhost:8000
```

Let me know if you think this design should move forward, or if I should revert the commit for less major accessibility changes that keep the current design. Thanks!

**Comment by jason-s-yu** [comment]
Sorry I accidentally dragged around some of your todo boxes under changes made/planned in your comment.

This is pretty good. I like the design. There's a couple small things we should probably adjust but I'm all for this rework. Thanks for your time.

Can you use a CDN for font serving (should be on Google fonts so you can just use their CDN)?

**Comment by devision1088** [comment]
We can use a CDN, I just don’t trust them because they have sometimes gone down in my past experience. However, a CDN is better for caching data. The font I used is not widely used, so it may not be cached from other websites already. I guess the choice is largely opinion based. What do you think?

**Comment by jason-s-yu** [comment]
The site operates directly from GitHub Pages. Should be ok to just host fonts in the repo.

**Comment by devision1088** [comment]
That should be fine, as the 100GB per month limit from GitHub Pages is very large.

@jason-s-yu Do you know any good ways to add parallax effects to block elements? I can't seem to find the best way that could be easily added without changing the HTML structure. For example, elements would scroll 50% slower while maintaining their spacing with a certain class.

parallax

**Comment by jason-s-yu** [comment]
[parallax.js](https://github.com/wagerfield/parallax)

**Comment by devision1088** [comment]
It appears that this responds to mouse position/device gyroscope. However, this would be great to implement anyways! I was looking for a solution that changes based on scroll, but I think that this can be done without a library if we translate the elements inside each element based on scroll position. Thanks!

**Comment by jason-s-yu** [comment]
[simpleparallax](https://simpleparallax.com/) will probably suit this project better

**Comment by devision1088** [comment]
Since the 3D canvas is displayed with a `<canvas>` element, [simpleparallax](https://simpleparallax.com) may not work because it needs to go on a `<section>` element that has children of `<canvas>` and `<article>`. I was looking at [Rellax](https://dixonandmoe.com/rellax/) since it seems to support any HTML element.

**Comment by jason-s-yu** [comment]
Sure, sounds good. Use whatever works best

**Comment by devision1088** [comment]
No library had what was needed, so I created a new one. [https://devision1088.com/unillax/](https://devision1088.com/unillax/). This should work on the website.

Also, what is the advantage of using [React](https://reactjs.org/)? Our website is a static one, so I do not think a virtual DOM would speed up page loading, and the library has a large file size (1.1 MB) which will slow loading. I have only used [React](https://reactjs.org/) in [React Native](https://facebook.github.io/react-native/), so I don't know much about the use of it in the web world :)

This is kind of off-topic, but would it be a good idea to make the website into a "web app" with [Google Firebase](https://firebase.google.com/)? This would allow people to have an account on the website (using [Firebase Authentication](https://firebase.google.com/products/auth/)) to save projects (with [Firebase Storage](https://firebase.google.com/products/storage/)) through an embedded editor on the website. This way, each person can write and save code in guided lessons on the website, eliminating the need for people copying and pasting, ultimately helping them develop better coding skills. This project may take a while, so don't worry about it.

Anyways, I'll hold off on any more changes as porting to React may take some work. The simpler the code to port, the better! (At least I hope :) )

**Comment by jason-s-yu** [comment]
Our website needs to host changing content so using react will allow us to integrate with a CMS more easily. Gzipped and bundled react site is ~100 kb which is small for js. 

Most likely will be using some static site generator like Gatsby which is highly optimized and provides us with near instantaneous page loads after the initial bundle is downloaded. Porting over to react will allow us to expand to put more microservices on the site if we decide to do it such as the code storage thing you are talking about.

Did you add any changes since #a232599/is this finalized and ready to merge?

**Comment by devision1088** [comment]
That makes sense. I'm going to close this since the changes I made were intended for a static website that pulled dynamic content through AJAX, which would be harder to port over to React.

**Comment by jason-s-yu** [comment]
No need. If your changes are ready for production let me know and I will merge

**Comment by devision1088** [comment]
What I meant was that to run locally, it needs to run on a server. It can be published to GitHub pages without any conflict. This is because I changed the file paths to look from the root, and running it locally would make it look for website files in the root of your computer, not the project folder. On a website, the project is hosted as root, meaning there will be no conflicts.

**_This, however, is not ready to be merged_**. If you clone and run it, you will see that about 0% of the old content was ported over to the new format. :) I'll try to port all the content over in the coming weeks, as I am getting busy. If you want to, feel free to commit to my branch and add any content.

If you want to run locally to see changes:
```shell
$ git clone https://github.com/devision1088/unics.club.git
$ cd unics.club
$ python3 -m http.server
```
``` shell
$ open http://localhost:8000
```

**Comment by devision1088** [comment]
@jason-s-yu The website does not require a server to run on. I meant that running the website locally for development purposes would require a local server on your computer.

> Our website needs to host changing content so using react will allow us to integrate with a CMS more easily. Gzipped and bundled react site is ~100 kb which is small for js.

Did you mean that the website would be static but use React to pull data from a CMS? Or did you mean that the website would be dynamically hosted in something like PHP or Node.js?

**Comment by devisionbit** [comment]
Sorry, I have not worked on this in a while (also the commits got messed up due to a username change). In summary, here's what has been changed so far:

- New design
- Full CSS markup for standard HTML 5 elements
- Switched development to a templated Node.js server using ejs.

The CMS that you mentioned earlier can be integrated in the back end instead of running on the front end to improve performance. If you want to see the current progress, install node locally on your machine, clone, and run `npm run start` from root or `npm run start-dev` for development updates.

I do now know what content you want on the website, so I have left sample text and elements in the pages to later integrate with the CMS. Templating can be found in the view folder and content can be passed to the views through the routes in the routes folder. Hope this helps!

**Comment by jason-s-yu** [comment]
@devisionbit thanks for your continued work. Hope you are staying safe in these times.

Is this site ready for merging?

**Comment by devisionbit** [comment]
Sorry for the mess here, I don't think I explained well. The site can be merged from commit 18c3058 or before, otherwise read below. DO NOT MERGE FROM THE LATEST COMMIT, it will break the website.

I switched the whole thing to a custom backend written in Node.js, so dynamic content could be pulled from a CMS in the back-end instead of relying on the client to do so. That probably was a bad idea since this site seems to be hosted on Github Pages, and I don't think GitHub Pages is meant to serve dynamic content. This can be hosted by any computer with Nodejs.

As of now, my GitHub account is very messed up, and due to privacy issues, I was forced to close my fork. You can take a look at my progress so far by cloning the branch and running `npm run start-dev`. The server will open on port 8080 of your machine.

I probably will not work on this project anymore since I have other projects at the moment, but I will explain the backend set up so you can finish it for the club's needs if you want. If the club does not need a dynamic website, you can merge up to commit 18c3058 (I think you are able to do this –I have given perms for you to commit to this branch). 

A quick summary of how the backend works: In the views folder, `.ejs` files are used to template the website. HTML includes can be found in the includes folder. These can be changed for your needs. In the routes folder, scripts are used to route and format ejs files to html files for the front-end. Templating from a cms can occur here. In app.js, the main requests are handled and route the request to the correct route script, then return the html file from the route script. The public folder contains static content hosted on the root of the site like css, scripts, and images. I have coded basic HTML layouts, and a full animated CSS stylesheet that works with all standard HTML elements (dropdowns, text entry fields, etc.)

So in summary: if you want to merge now, ONLY MERGE UP TO 18c3058!! All commits after are for a dynamic custom back-end server. The basic layout of the server backend has been written, and you can tweak and finish it for the club's needs if you want. Otherwise, merge up to 18c3058.

I hope that helps! Stay safe! Responses may be slow, I am working heavily on other projects at the moment.
